src/system/arena-data.ts

```diff
@@ -27,7 +28,7 @@ export default class ArenaData {
     this.tags = [];
 
     if (source.tags) {
-      this.tags = source.tags.map((t) => loadArenaTag(t));
+      this.tags = source.tags.map((t) => loadArenaTag(t)).filter((t) => !isNil(t));
```

Comment:
You should tell the compiler here that you know it's an array of ArenaTags 

```suggestion
      this.tags = (source.tags as ArenaTag[]).map((t) => loadArenaTag(t)).filter((t) => !isNil(t));
```